08.ls_object/lib/ls.rb

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+require_relative 'display'
+
+class Ls
+  attr_reader :display
+
+  def ls
+    options = parse_options
+    target = ARGV[0] || '.'
+
+    raise(ArgumentError, "ls: #{target} is no such file or directory") unless File.exist?(target)
```

Comment:
```suggestion
    unless File.exist?(target)
      raise(ArgumentError, "ls: #{target} is no such file or directory") 
    end
```
メッセージの組み立てが長くなっていたので、普通にブロックのほうがわかりやすそうです。

Reply:
そうだなぁと思って試してみたんですが…… 👮🏻‍♂️ <「unless修飾子のほうがいいよ」と怒られてしまうんですよぇ

```ruby
    if File.exist?(target)
      @display = Display.new(target)

      options.key?(:a) ? display.list_contain_dotfile : display.list_without_dotfile
      options.key?(:r) ? display.list_reverse : display.list_sort
      options.key?(:l) ? display.long_format : display.short_format
    else
      raise(ArgumentError, "ls: #{target} is no such file or directory")
    end
```

としたら、今度は「ガード節を使い給へ」と指摘されました 🤔